lib/stdlib/test/property_test/queue_prop.erl

```diff
@@ -0,0 +1,670 @@
+%%
+%% %CopyrightBegin%
+%%
+%% Copyright Ericsson AB 2021. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%
+%% %CopyrightEnd%
+%%
+-module(queue_prop).
+
+-compile(export_all).
+
+-proptest(eqc).
+-proptest([triq, proper]).
+
+-ifndef(EQC).
+-ifndef(PROPER).
+-ifndef(TRIQ).
+-define(EQC, true).
+-endif.
+-endif.
+-endif.
+
+-ifdef(EQC).
+-include_lib("eqc/include/eqc.hrl").
+-define(MOD_eqc,eqc).
+
+-else.
+-ifdef(PROPER).
+-include_lib("proper/include/proper.hrl").
+-define(MOD_eqc,proper).
+
+-else.
+-ifdef(TRIQ).
+-define(MOD_eqc,triq).
+-include_lib("triq/include/triq.hrl").
+
+-endif.
+-endif.
+-endif.
+
+%%%%%%%%%%%%%%%%%%
+%%% Properties %%%
+%%%%%%%%%%%%%%%%%%
+
+prop_new() ->
+    [] =:= queue:to_list(queue:new()).
+
+prop_is_queue() ->
+    ?FORALL(
+	{IsQueue, Q},
+	oneof([
+	    {true, queue()},
+	    {false, non_queue()}
```

Comment:
Considering the implementation of the `non_queue/0` generator, this will/might start to fail when/if the internal representation of queues change.  While this is very unlikely to happen, it does show the problem of the `is_queue/1` function, i.e., it can return false positives.
On the other hand, this is not to say that this is a bad test. The `true` part of it tests that what is generated by the `queue/0` generator is considered to be a queue, which is good.  The `false` part is more questionable since it relies on knowledge of representation.  It does, however, have sufficient value to keep it.

Reply:
I don't think that relying on the internal representation is too bad here since where we are testing the module itself, not something that _relies_ on it. If the internal representation changes, many other things are bound to change besides the tests, anyway, right?

Reply:
See previous comment about the `non_queue/0` generator.  Again, adding some documentation to explain the brittleness might be good for future readers.
Also, this also shows the difference between _internal_ (concrete) tests and _external_ (abstract) tests.